cpp/foxglove/include/foxglove/error.hpp

```diff
@@ -0,0 +1,38 @@
+
+#include <exception>
+#include <memory>
+
+struct foxglove_error;
+
+namespace foxglove {
+
+enum class FoxgloveErrorKind {
+  Unspecified,
+  ValueError,
+  SinkClosed,
+  SchemaRequired,
+  MessageEncodingRequired,
+  ServerAlreadyStarted,
+  Bind,
+  DuplicateChannel,
+  DuplicateService,
+  MissingRequestEncoding,
+  ServicesNotSupported,
+  ConnectionGraphNotSupported,
+  IoError,
+  McapError,
+};
+
+class FoxgloveError : public std::exception {
```

Comment:
I don't think this really matters much, just a habit I've ended up with to avoid surprises
```suggestion
class FoxgloveError : public std::exception final {
```